src/Tests/Addressing/NamespacePartitioning/When_using_sharded_strategy_on_multiple_namespaces.cs

```diff
@@ -68,26 +68,27 @@ public void Sharded_partitioning_strategy_will_circle__the_active_namespace()
         {
             var i = 0;
 
-            var buckets = new List<String>()
+            var buckets = new Dictionary<string, string>()
             {
-                "Endpoint=sb://namespace1.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=somesecretkey",
-                "Endpoint=sb://namespace2.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=somesecretkey",
-                "Endpoint=sb://namespace3.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=somesecretkey"
+                { "name1", "Endpoint=sb://namespace1.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=somesecretkey"},
+                { "name2", "Endpoint=sb://namespace2.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=somesecretkey"},
+                { "name3", "Endpoint=sb://namespace3.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=somesecretkey"}
             };
 
             var settings = new SettingsHolder();
             var extensions = new TransportExtensions<AzureServiceBusTransport>(settings);
-            foreach (var s in buckets)
+            foreach (var bucket in buckets)
             {
-                extensions.UseDefaultTopology().Addressing().NamespacePartitioning().AddNamespace(s);
+                extensions.UseDefaultTopology().Addressing().NamespacePartitioning().AddNamespace(bucket.Key, bucket.Value);
             }
 
             var strategy = new ShardedNamespacePartitioningStrategy(settings);
             strategy.SetShardingRule(() => i);
-            
+
             for (i = 0; i < 3; i++)
```

Comment:
I'm probably the one to be blamed for this - should we switch to collection.Count here?